### PR TITLE
Add basic perf record structure and reader for tracepoints

### PR DESCRIPTION
**Pull request by ronaldfw**

### CODE REVIEW

File: OrbitLinuxTracing/PerfEventReaders.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#include "PerfEventReaders.h"
+
+#include <OrbitBase/Logging.h>
+#include "PerfEventRingBuffer.h"
+
+namespace LinuxTracing {
+
+uint16_t ReadTracepointCommonType(PerfEventRingBuffer* ring_buffer) {
```
Comment (dpallotti) [inline OrbitLinuxTracing/PerfEventReaders.cpp]: I can see that you like functions being defined in a cpp file, but is it that necessary to do differently than all the other functions here? It makes this one look special. It's even a good candidate for inlining...

File: OrbitLinuxTracing/PerfEventRecords.h [hunk 2]
```diff
@@ -86,6 +86,21 @@ struct __attribute__((__packed__)) perf_event_lost {
...
+// record and the common type. The common type is the same as the tracepoint id
+// and we use it to identify the events and how to handle them. The actual
+// record is thus larger than the size of this struct but since it is dynamic
+// and depends on the type of the tracepoint, we only hardcode the common part
+// here.
+struct __attribute__((__packed__)) perf_event_tracepoint {
+  perf_event_header header;
+  perf_event_sample_id_tid_time_streamid_cpu sample_id;
+  uint32_t size;
+  uint16_t common_type;
```
Comment (dpallotti) [inline OrbitLinuxTracing/PerfEventRecords.h]: So this is already part of 
`char  data[size];   /* if PERF_SAMPLE_RAW */`
right? Can you specify that with a comment?

File: OrbitLinuxTracing/PerfEventRecords.h [hunk 3]
```diff
@@ -86,6 +86,21 @@ struct __attribute__((__packed__)) perf_event_lost {
...
 
+// Tracepoints are perf samples, so they start with the same layout as a regular
+// perf sample, that is, with the header and the common fields like sample_id,
+// tid, timestamp, etc. The next field is the size of the rest of the tracepoint
+// record and the common type. The common type is the same as the tracepoint id
+// and we use it to identify the events and how to handle them. The actual
+// record is thus larger than the size of this struct but since it is dynamic
+// and depends on the type of the tracepoint, we only hardcode the common part
+// here.
+struct __attribute__((__packed__)) perf_event_tracepoint {
```
Comment (dpallotti) [inline OrbitLinuxTracing/PerfEventRecords.h]: If this is only a part of the sample, wouldn't it be better if its name reflected it? Something along the lines of perf_event_tracepoint_common?

File: OrbitLinuxTracing/PerfEventReaders.cpp (hunk 1 above)
Comment (ronaldfw) [reply OrbitLinuxTracing/PerfEventReaders.cpp]: I actually think we should move them all to a cpp file. There's no reason for inlining them and it's easier to debug if they are defined in a cpp file. I can take care of moving them.

File: OrbitLinuxTracing/PerfEventRecords.h (hunk 3 above)
Comment (ronaldfw) [reply OrbitLinuxTracing/PerfEventRecords.h]: Done

File: OrbitLinuxTracing/PerfEventRecords.h (hunk 2 above)
Comment (ronaldfw) [reply OrbitLinuxTracing/PerfEventRecords.h]: Done.

File: OrbitLinuxTracing/PerfEventReaders.cpp (hunk 1 above)
Comment (ronaldfw) [reply OrbitLinuxTracing/PerfEventReaders.cpp]: Moved all functions to the cpp file.

File: OrbitLinuxTracing/PerfEventRecords.h [hunk 4]
```diff
@@ -86,6 +86,25 @@ struct __attribute__((__packed__)) perf_event_lost {
...
+// and we use it to identify the events and how to handle them. The actual
+// record is thus larger than the size of this struct but since it is dynamic
+// and depends on the type of the tracepoint, we only hardcode the common part
+// here.
+struct __attribute__((__packed__)) perf_event_tracepoint_common {
+  perf_event_header header;
+  perf_event_sample_id_tid_time_streamid_cpu sample_id;
+  uint32_t size;
+  // The rest of the struct is a char[size], but all tracepoints record the
+  // tracepoint id in the first two types. Since we need to identify the
```
Comment (dpallotti) [inline OrbitLinuxTracing/PerfEventRecords.h]: "first two types"?

File: OrbitLinuxTracing/PerfEventReaders.cpp (hunk 1 above)
Comment (dpallotti) [reply OrbitLinuxTracing/PerfEventReaders.cpp]: Alright then.

Comment (ronaldfw) [reply OrbitLinuxTracing/PerfEventReaders.cpp]: I did a quick benchmark using "perf record" on OrbitService while instrumenting the top ~20 functions and none of the read functions here show up in the profile at all. Do you have any data that indicates these should be inlined?

**Pull request merged**